pkg/templateservicebroker/api/types.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// +genclient=true
+
+type TemplateProvision struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec   TemplateProvisionSpec
+	Status TemplateProvisionStatus
+}
+
+type TemplateProvisionSpec struct {
+	Template  templateapi.Template
+	SecretRef kapi.LocalObjectReference
+}
+
+type TemplateProvisionStatus struct {
+	State TemplateProvisionState
+}
+
+type TemplateProvisionState string
+
+const (
+	TemplateProvisionStateNew      TemplateProvisionState = "New"
+	TemplateProvisionStateCreating TemplateProvisionState = "Creating"
+	TemplateProvisionStateCreated  TemplateProvisionState = "Created"
+	TemplateProvisionStateFailed   TemplateProvisionState = "Failed"
+)
+
+// TemplateList is a list of Template objects.
+type TemplateProvisionList struct {
+	unversioned.TypeMeta
+	unversioned.ListMeta
+
+	Items []TemplateProvision
+}
+
+type UUIDMap struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec UUIDMapSpec
+}
+
+type UUIDMapSpec struct {
+	TemplateProvisionRef kapi.ObjectReference
```

Comment:
sounds like this should be a TemplateServiceUUIDMapSpec then.  (TemplateService because what we are creating from a provision request is, ostensibly, a service instance of type template)